src/components/CTABlock.tsx

```diff
@@ -0,0 +1,49 @@
+import { useState, useCallback } from 'react'
+import { EventControlledContent } from './EventControlledContent'
+import { mathifyElement } from '../lib/math'
+
+interface CTABlockProps {
+  contentString: string
+  contentPrompt: string
+  buttonText: string
+  fireEvent?: string
+  waitForEvent?: string
+
+}
+
+export const CTABlock = ({ contentString, contentPrompt, buttonText, fireEvent, waitForEvent }: CTABlockProps): JSX.Element => {
+  const [clicked, setclicked] = useState<boolean>(false)
+
+  const clickHandler = (): void => {
+    if (fireEvent !== undefined) {
+      const clickEvent = new CustomEvent(fireEvent)
+      document.dispatchEvent(clickEvent)
+    }
+    setclicked(true)
+  }
+
+  const contentRefCallback = useCallback((node: HTMLDivElement | null): void => {
+    if (node != null) {
+      mathifyElement(node)
```

Comment:
Leaving a corresponding note as a marker for [this comment](https://github.com/openstax/k12-apps-raise/pull/8#discussion_r863923597) here.